src/Spice86/Behaviors/DisassemblyScrollBehavior.cs

```diff
@@ -0,0 +1,243 @@
+    }
+
+    // Helper methods for getting/setting the attached properties
+    public static bool GetIsEnabled(Control control) {
+        return control.GetValue(IsEnabledProperty);
+    }
+
+    public static void SetIsEnabled(Control control, bool value) {
+        control.SetValue(IsEnabledProperty, value);
+    }
+
+    public static SegmentedAddress GetTargetAddress(Control control) {
+        return control.GetValue(TargetAddressProperty);
+    }
+
+    public static void SetTargetAddress(Control control, SegmentedAddress value) {
+        control.SetValue(TargetAddressProperty, value);
+    }
+
+    // Handler for when IsEnabled property changes
+    private static void OnIsEnabledChanged(Control control, AvaloniaPropertyChangedEventArgs e) {
+        if (control is not ListBox listBox) {
+            return;
+        }
+
+        bool isEnabled = (bool)e.NewValue!;
+        if (isEnabled) {
+            // Subscribe to the Loaded event to handle initial centering
+            listBox.Loaded += OnListBoxLoaded;
+
+            // If the ListBox is already loaded, check if we need to scroll
+            if (listBox.IsLoaded) {
+                uint targetAddress = GetTargetAddress(listBox).Linear;
+                if (targetAddress != 0) {
+                    // Delay the scroll to ensure the UI has fully loaded
+                    Dispatcher.UIThread.Post(() => ScrollToAddress(listBox, targetAddress), DispatcherPriority.Loaded);
+                }
+            }
+        } else {
+            listBox.Loaded -= OnListBoxLoaded;
+        }
+    }
+
+    // Handler for ListBox Loaded event
+    private static void OnListBoxLoaded(object? sender, EventArgs e) {
+        if (sender is not ListBox listBox) {
+            return;
+        }
+
+        // Unsubscribe from the Loaded event to avoid multiple calls
+        listBox.Loaded -= OnListBoxLoaded;
+
+        // Check if we have a target address for this ListBox
+        uint targetAddress = GetTargetAddress(listBox).Linear;
+        if (targetAddress != 0) {
+            Dispatcher.UIThread.Post(() => ScrollToAddress(listBox, targetAddress), DispatcherPriority.Loaded);
+        }
+    }
+
+    // Handler for when TargetAddress property changes
+    private static void OnTargetAddressChanged(Control control, AvaloniaPropertyChangedEventArgs e) {
+        if (control is not ListBox listBox || !GetIsEnabled(listBox)) {
+            return;
+        }
+        uint targetAddress = ((SegmentedAddress)e.NewValue!).Linear;
+        if (listBox.IsLoaded) {
+            ScrollToAddress(listBox, targetAddress);
+        } else {
+            listBox.Loaded -= OnListBoxLoaded;
+            listBox.Loaded += OnListBoxLoaded;
+        }
+    }
+
+    // Method to scroll to a specific address in the disassembly view
+    public static void ScrollToAddress(ListBox listBox, uint targetAddress) {
+        // Ensure we're on the UI thread
+        if (!Dispatcher.UIThread.CheckAccess()) {
```

Comment:
Function doesnt seem to be very CPU intensive, is it worth dispatching it on a UI thread? Is it dispatched for another reason like access to avalonia structures? (Sorry, I almost never touched avalonia)

Reply:
This does the scrolling animation of the ListBox. All user-interface manipulation should be done on the UI thread.